map/map_loader/src/lanelet2_map_loader/lanelet2_differential_loader_module.cpp

```diff
@@ -0,0 +1,216 @@
+// Copyright 2024 The Autoware Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "map_loader/lanelet2_differential_loader_module.hpp"
+
+#include "lanelet2_local_projector.hpp"
+#include "map_loader/lanelet2_map_loader_node.hpp"
+
+Lanelet2DifferentialLoaderModule::Lanelet2DifferentialLoaderModule(
+  rclcpp::Node * node,
+  const std::map<std::string, Lanelet2FileMetaData> & lanelet2_file_metadata_dict, double & x_res,
+  double & y_res, const double & center_line_resolution)
+: logger_(node->get_logger()),
+  clock_(node->get_clock()),
+  lanelet2_file_metadata_dict_(lanelet2_file_metadata_dict),
+  center_line_resolution_(center_line_resolution)
+{
+  pub_whole_map_bin_ =
+    node->create_publisher<LaneletMapBin>("output/lanelet2_map", rclcpp::QoS{1}.transient_local());
+
+  const auto metadata_adaptor = component_interface_utils::NodeAdaptor(node);
+  metadata_adaptor.init_pub(pub_lanelet_map_meta_data_);
+
+  const auto adaptor = component_interface_utils::NodeAdaptor(node);
+  adaptor.init_sub(
+    sub_map_projector_info_,
+    [this, &x_res, &y_res](const map_interface::MapProjectorInfo::Message::ConstSharedPtr msg) {
+      projector_info_ = *msg;
+
+      // load whole map once when projector info is available
+      {
+        const auto map = loadWholeMap();
+        pub_whole_map_bin_->publish(map);
```

Comment:
I don't want Lanelet2DifferentialLoaderModule to publish the entire map or metadata.
This was originally done by Lanelet2MapLoaderNode, so please modify it to delegate these tasks to Lanelet2MapLoaderNode.

Reply:
if we can solve the previous review (https://github.com/autowarefoundation/autoware.universe/pull/6241#discussion_r1671985468) somehow, we can find a good solution for this one.

Reply:
No, we have a code block in `lanelet2_map_loader_node.cpp` from line 161 to 167 to load whole map for once.